src/test/ui/rfc-2632-const-trait-impl/issue-100222.rs

```diff
@@ -0,0 +1,14 @@
+// check-pass
+#![feature(const_trait_impl)]
+
+#[const_trait]
+pub trait Index {
+    type Output;
+}
+
+#[const_trait]
+pub trait IndexMut where Self: Index {
+    fn foo(&mut self) -> <Self as Index>::Output;
+}
```

Comment:
So... the problem was that this is not `Self: ~const Index`, but it was tested as such? Do we have tests for various combinations of `#[const_trait]` being on the super trait or not, and being on the trait itself or not?